aggregateur/validators.py

```diff
@@ -342,3 +342,75 @@ def schema_json_data(self):
         with open(self.filepath(self.SCHEMA_FILENAME)) as f:
             self.schema_data = json.load(f)
         return self.schema_data
+
+
+class GenericValidator(BaseValidator):
+    SCHEMA_FILENAME = "schema.yml"
+
+    def __init__(self, repo):
+        super().__init__(repo)
+        self.schema_data = None
+        self.has_changelog = False
+
+    def schemas_metadata(self):
+        return [
+            {
+                "path": self.SCHEMA_FILENAME,
+                "original_path": self.SCHEMA_FILENAME,
+                "title": self.title,
+                "latest_url": self.schema_url(self.SCHEMA_FILENAME),
+            }
+        ]
+
+    @property
+    def title(self):
+        return self.get_schema_data()["title"]
+
+    @property
+    def description(self):
+        return self.get_schema_data()["description"]
+
+    @property
+    def homepage(self):
+        return self.get_schema_data()["homepage"]
+
+    def validate(self):
+        super().validate()
+        self.check_file_exists(self.SCHEMA_FILENAME)
+        self.check_schema(self.SCHEMA_FILENAME)
+        self.check_extra_keys()
+
+    def extract(self):
+        files = {
+            self.SCHEMA_FILENAME: self.filepath_or_none(self.SCHEMA_FILENAME),
+            "README.md": self.filepath_or_none("README.md"),
+            "SEE_ALSO.md": self.filepath_or_none("SEE_ALSO.md"),
+            "CONTEXT.md": self.filepath_or_none("CONTEXT.md"),
+        }
+
+        if self.is_latest_version():
+            files[self.CHANGELOG_FILENAME] = self.filepath_or_none(
+                self.CHANGELOG_FILENAME
+            )
+
+        self.move_files(files)
+
+    def check_extra_keys(self):
+        keys = ["title", "description", "homepage", "version"]
+        for key in [k for k in keys if k not in self.get_schema_data()]:
+            message = "Key `%s` is a required key and is missing from %s" % (
+                key,
+                self.SCHEMA_FILENAME,
+            )
+            raise exceptions.InvalidSchemaException(self.repo, message)
+
+    def check_schema(self, filename):
+        return True
```

Comment:
Consider deleting this method.

Reply:
Did a little Yaml dance instead.

Reply:
BTW what happens when one of the tags is invalid? The other tags are still published on schema.data.gouv.fr, right?

Reply:
Yup, only valid tags are published. If there are errors you'd get an email with details

https://github.com/etalab/schema.data.gouv.fr/blob/56f8207f61d1a153bdbe98c081ab9ad7f1e827c4/aggregateur/main.py#L276-L284